### ISSUE TITLE
 why updateType = 'weight'?

### ISSUE THREAD

**Issue by bolianchen** [xiangning-chen/SmoothDARTS#1]
Hi, would you please explain why you bypass softmax in the forward computation while applying projected gradient descent to maximize the loss wrt arch_parameters? How is the performance compared to updateType='alpha' 

Thanks!

**Comment by bolianchen** [comment]
I found you use updateType='weight' everywhere. It seems the design is to directly learn weights of the mixed operations
-    softmax_arch_parameters is called once to initialize arch_parameters as probability-like
-    model.clip() is used to maintain the probability-like property

but I am not pretty sure about the benefits of doing it?

**Comment by bolianchen** [comment]
another question, if I go with updateType='alpha', model.clip() seems unnecessary any more. am I correct?

**Comment by xiangning-chen** [comment]
Thanks for your question, this parameter is for performing perturbation. We perform perturbation after softmax and then we use the perturbed weight directly.

**Comment by xiangning-chen** [comment]
When updating alpha in the architecture, we set the parameter as 'alpha'.
